src/test/java/spoon/test/prettyprinter/TestSniperPrinter.java

```diff
@@ -876,6 +876,69 @@ void test_retainSemiColonAfterTheLastResource() {
 		}
 	}
 
+	@Nested
+	@GitHubIssue(issueNumber = 4315)
+	class SquareBracketPrintingInArrayInitialisation {
+		// contract: square brackets should be printed *only* after the identifier of the field or local variable
```

Comment:
@monperrus note that I have added the contract for the nested case instead of individual test cases because all of them are testing the same thing but resources are different.